bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,213 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.energenie.internal.config.EnergenieConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergeniePWMHandler} is responsible for reading states and update PWM channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+
+@NonNullByDefault
+public class EnergeniePWMHandler extends BaseThingHandler {
+
+    public enum PWMState {
+        VOLTAGE("var V  = ", 9, 20, 10, SmartHomeUnits.VOLT),
+        CURRENT("var V  = ", 9, 20, 100, SmartHomeUnits.AMPERE),
+        POWER("var P=", 6, 20, 466, SmartHomeUnits.WATT),
+        ENERGY("var E=", 6, 20, 25600, SmartHomeUnits.WATT_HOUR);
+
+        private final Logger logger = LoggerFactory.getLogger(PWMState.class);
+
+        private final String stateResponseSearch;
+        private final int start;
+        private final int stop;
+        private final int divisor;
+        private final Unit<?> unit;
+
+        private PWMState(final String stateResponseSearch, final int start, final int stop, final int divisor,
+                final Unit<?> unit) {
+            this.stateResponseSearch = stateResponseSearch;
+            this.start = start;
+            this.stop = stop;
+            this.divisor = divisor;
+            this.unit = unit;
+        }
+
+        public String getStateResponseSearch() {
+            return stateResponseSearch;
+        }
+
+        public int getStart() {
+            return start;
+        }
+
+        public int getStop() {
+            return stop;
+        }
+
+        public int getDivisor() {
+            return divisor;
+        }
+
+        public State readState(final String loginResponseString) {
+            final int findState = loginResponseString.lastIndexOf(stateResponseSearch);
+
+            if (findState > 0) {
+                logger.trace("searchstring {} found at position {}", stateResponseSearch, findState);
+                final String slicedResponseTmp = loginResponseString.substring(findState + start, findState + stop);
+                logger.trace("transformed state response = {}", slicedResponseTmp);
+                final String[] slicedResponse = slicedResponseTmp.split(";");
+                logger.trace("transformed state response = {} - {}", slicedResponse[0], slicedResponse[1]);
+                final double value;
+
+                try {
+                    if (Double.parseDouble(slicedResponse[0]) / 1 == Double.parseDouble(slicedResponse[0])) {
```

Comment:
I'm confused what you are trying to check here. `n/1 == n`? Isn't that always true?

Reply:
@chris-kuhr can you explain please, as you introduced this code to the openHAB 1.x version.

Reply:
Thinking a bit more about it, you are right. This division will always return true, therefore I will remove it.

Reply:
Done